addons/sale/migrations/9.0.1.0/post-migration.py

```diff
@@ -100,10 +101,12 @@ def set_so_line_computed_rest(env):
         FROM (
             SELECT rel.order_line_id,
                 SUM(CASE
-                    WHEN ai.type = 'out_invoice'
+                    WHEN ai.type = 'out_invoice' AND u.category_id = u2.category_id
```

Comment:
This is not something reflected on ORM code and trying to prevent an error that should raise any way in other places (incompatible UoMs).

Reply:
The ORM has this code:

https://github.com/OCA/OpenUpgrade/blob/bd1a1d7b04b1d5ea5d94b931443d61fc71d810a2/addons/product/product.py#L120-L124

It contemplates a situation where it's possible for incompatible UoM categories to land.

Indeed this context key that is required wouldn't exist during migration and it would fail instead. In any case we were losing also the exception with previous implementation.

So do I leave it like it is, or do I add the check and raise an exception when incompatible UoMs are found? (I'd have to search for them via SQL).